gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -133,30 +133,44 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
         {%- endif %}
         """
 
-        {%- if 'body' in method.http_opt.keys() %}
+        {%- if 'body' in method.http_opt %}
         # Jsonify the input
-        data = {{ method.output.ident }}.to_json(
-            {%- if method.http_opt['body'] == '*' %}
+        {%- if method.http_opt['body'] != '*' %}
+        data = {{ method.input.fields[method.http_opt['body']].type.ident }}.to_json(
+            request.{{ method.http_opt['body'] }},
+            including_default_value_fields=False
+        )
+        {%- else %}
+        data = {{ method.input.ident }}.to_json(
             request
-            {%- else %}
-            request.body
-            {%- endif %}
         )
         {%- endif %}
+        {%- endif %}
 
         {# TODO(yon-mg): Write helper method for handling grpc transcoding url #}
         # TODO(yon-mg): need to handle grpc transcoding and parse url correctly
-        #              current impl assumes simpler version of grpc transcoding
+        #               current impl assumes basic case of grpc transcoding
         # Send the request
         url = 'https://{host}{{ method.http_opt['url'] }}'.format(
             host=self._host,
-            {%- for field in method.input.fields.keys() %}
+            {%- for field in method.path_params %}
             {{ field }}=request.{{ field }},
             {%- endfor %}
         )
+
+        potentialParams = {
+            {%- for field in method.query_params %}
+            '{{ field|camel_case }}': request.{{ field }},
+            {%- endfor %}
+        }
+        potentialParams = ((k, v) for k, v in potentialParams.items() if v)
```

Comment:
!! re: `if v`
Is the Python truthiness of `v` what we want here? I don't know, but I wonder given that we want to transmit only explicitly-set values, even if they match the defaults, which are falsy.

For any data type in the Python data model:
- if the unset value is null, then we can test for null to determine whether to send it. Simply checking for falsiness would not make this distinction.
- if the unset value is non-null, is there some other way to distinguish between values that were explicitly set and those that weren't?

We should talk about this, and at the very least insert a TODO for now.

Reply:
Short answer is yes and your second bullet point applies. Afaik from debugging it, there isn't actually a way to tell if a value was unset or set to default value since when the `<InputType>Request` object is made and passed to this generated method as a parameter, its values seem set to default ones. I also believe this is intended. For example, if I made a request but set a query param value to an empty string, simply checking if a value was set would allow that empty string to be included as part of the path_url and may result in a `'Bad Request'` response.

Reply:
OK, this ties back to the default values design I still have open.....

On empty strings, maybe  one of the [ideas from the web](https://www.google.com/search?q=empty+strings+as+query+params&oq=empty+strings+as+query+params&aqs=chrome..69i57j69i64l3.6119j1j7&sourceid=chrome&ie=UTF-8) will work with OP APIs. But we probably don't need to spend much time investigating until we decide on a the approach to default values. It certainly shouldn't block this PR.

But yes, let's add a TODO that likely something will need to be done here once we decide what we're doing for default values.